helm/linkerd2-app/templates/identity.yaml

```diff
@@ -89,8 +114,31 @@ spec:
       containers:
       - args:
         - identity
-        - -log-level={{.Values.controllerLogLevel}}
+        - -log-level={{.Values.global.controllerLogLevel}}
+        - -controller-namespace={{.Release.Namespace}}
+        - -identity-trust-domain={{.Values.global.identityTrustDomain}}
+        - -identity-issuance-lifetime={{.Values.identity.issuer.issuanceLifetime}}
+        - -identity-clock-skew-allowance={{.Values.identity.issuer.clockSkewAllowance}}
+        {{- if and (.Values.global.identity.issuer) (eq .Values.global.identity.issuer.scheme "linkerd.io/tls") -}}
+        {{/* use certificate provided in values */}}
+        - -identity-trust-anchors-pem={{b64enc .Values.global.identityTrustAnchorsPEM}}
+        {{- end }}
+        {{- if and (.Values.global.identity.issuer) (eq .Values.global.identity.issuer.scheme "kubernetes.io/tls") -}}
+        {{/* use certificates created by cert-manager */}}
+        - -identity-trust-anchors-pem=$(ROOT_CA)
+        - -issuer=/var/run/linkerd/identity/issuer
+        {{- end }}
+        - -identity-scheme={{.Values.identity.issuer.scheme}}
         {{- include "partials.linkerd.trace" . | nindent 8 -}}
+        {{- if and (.Values.global.identity.issuer) (eq .Values.global.identity.issuer.scheme "kubernetes.io/tls") -}}
+        {{/* inject the root ca (trust anchor) which was converted to a base64 encoded string by the pre-install hook */}}
+        env:
+          - name: ROOT_CA
+            valueFrom:
+              secretKeyRef:
+                name: linkerd-trust-anchor-base64
+                key: ca.crt.base64
+        {{- end }}
```

Comment:
use the certs generated during the pre-install hook to create the deployment. this mounts the double-encoded root ca and then passes it as an env var to the identity container.